File: Day2/Day2.py
import os

cwd = os.getcwd()  # Get the current working directory (cwd)
files = os.listdir(cwd)  # Get all the files in that directory
print("Files in %r: %s" % (cwd, files))


horizontal = 0
vertical = 0
aim = 0
depth = 0

with open("Day2/problem_input.txt") as file:
    for line in file:
        direction, quantity = line.split(" ")
        quantity = int(quantity)
        if direction == "forward":
            horizontal += quantity
            depth += quantity * aim
        elif direction == "up":
            aim -= quantity
        elif direction == "down":
            aim += quantity

product = depth * horizontal
print(product) # 1580000 for part 1
# 1251263225 for part 2

File: Day2/Day2.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
using namespace std;


void part1() {
    fstream myfile("problem_input.txt", std::ios_base::in);

    // vector<string> commands;
    string command, delim = " ", direction, quantity;
    int horizontal = 0, vertical = 0;
    
    while (myfile >> command)
    {
        // commands.push_back(command);
        direction = command.substr(0, command.find(delim));
        quantity = command.substr(command.find(delim) + 1, command.length());
     
        // forward, up, down
        if (direction == "forward") {
            horizontal += stoi(quantity);
        } else if (direction == "up") {
            vertical -= stoi(quantity);
        } else if (direction == "down") {
            vertical += stoi(quantity);
        }
    }

    int product = horizontal * vertical;
    cout << "Product: " << product << endl;
}

void part2() {
    fstream myfile("problem_input.txt", std::ios_base::in);

    vector<int> depths;
    int depth, index = -1, num_inc = 0, window1, window2;

    
}

int main(int argc, char * argv[])
{
    part1(); // 1754
    // part2(); // 1789
}

